extensions/wikibase/src/org/openrefine/wikibase/editing/MediaFileUtils.java

```diff
@@ -102,6 +106,62 @@ public MediaUploadResponse uploadLocalFile(File path, String fileName, String wi
         return uploadFile(parameters, files);
     }
 
+    /**
+     * Upload a local file to the MediaWiki instance in chunks.
+     * 
+     * @param path
+     *            ChunkedFile of the local file
+     * @param fileName
+     *            its filename once stored on the wiki
+     * @param wikitext
+     *            the accompanying wikitext for the file
+     * @param summary
+     *            the edit summary associated with the upload
+     * @param tags
+     *            tags to apply to the edit
+     * @return
+     * @throws IOException
+     * @throws MediaWikiApiErrorException
+     */
+    protected MediaUploadResponse uploadLocalFileChunked(ChunkedFile path, String fileName, String wikitext, String summary,
+            List<String> tags)
+            throws IOException, MediaWikiApiErrorException {
+        MediaUploadResponse response = null;
+        int i = 1;
+        for (File chunk = path.readChunk(); chunk != null; chunk = path.readChunk()) {
+            Map<String, String> parameters = new HashMap<>();
+            parameters.put("action", "upload");
+            parameters.put("token", getCsrfToken());
+            parameters.put("stash", "1");
+            parameters.put("filename", fileName);
+            parameters.put("filesize", String.valueOf(path.getLength()));
+            if (response == null) {
+                // In the first request we don't have offset or file key.
+                parameters.put("offset", "0");
+            } else {
+                parameters.put("offset", String.valueOf(response.offset));
+                parameters.put("filekey", response.filekey);
+            }
+            Map<String, ImmutablePair<String, java.io.File>> files = new HashMap<>();
+            String chunkName = "chunk-" + i + ".png";
+            files.put("chunk", new ImmutablePair<String, File>(chunkName, chunk));
+            response = uploadFile(parameters, files);
```

Comment:
After calling `uploadFile` it's probably worth calling `response.checkForErrors()` so that we can abort the upload early if uploading any of the chunk fails.